Automattic/pocket-casts-ios
# [Give Ratings] API implementation

Fixes #1884

This PR implement the 2 APIs toget and submit a podcast rating. The APIs are:
• `/user/podcast_rating/add` to update a podcast that has previous rate
• `/user/podcast_rating/show` to fetch the individual rate
The APIs are implement using 2 new tasks based on the Protobuf implementation.
This PR also adds
• hide the Submit button when the the previous rate and current rate selected to update is the same
• update the dismiss button position based on Figma design

## To test

> [!NOTE]
> Have `giveRatings` feature flag enabled
> Please use Staging to test

1. Open a podcast that you did not listen any episode
2. Try to rate a podcast that you did not rate before
3. ✅ Ensure the rating screen is opened with submit button disabled
4. Select any rate
5. ✅ Ensure the submit button is enabled
6. Rate the podcast
7. Open the rating screen again
8. ✅ Ensure you see the rate you gave in step 6 and submit button is hidden
9. Select a different rate
10. ✅ Ensure submit button is visible
11. Disable the feature flag
12. ✅ Test everything still works as normal

## Recording

https://github.com/Automattic/pocket-casts-ios/assets/912252/fa70d2c7-a17e-46a2-82a2-73d66c71fcbb


## Checklist
